agimus_controller/mpc.py

```diff
@@ -1,380 +1,326 @@
-        else:
-            u_reg_cost = self.get_control_residual(np.zeros([self.nq, 1]))
-            running_cost_model.addCost("uReg", u_reg_cost, 0)
-        return crocoddyl.IntegratedActionModelEuler(
-            crocoddyl.DifferentialActionModelFreeFwdDynamics(
-                self.state, self.actuation, running_cost_model
-            ),
-            self.DT,
-        )
-
-    def get_last_model_with_mim(self):
-        """Return last model for a sub path with constraints for mim_solvers."""
-        running_cost_model = crocoddyl.CostModelSum(self.state)
-        constraints = crocoddyl.ConstraintModelManager(self.state, self.nv)
-
-        # add placement constraint
-        placement_residual = self.get_placement_residual(self.x_plan[-1, : self.nq])
-        placement_constraint = crocoddyl.ConstraintModelResidual(
-            self.state,
-            placement_residual,
-            np.array([0] * 12),
-            np.array([1e-3] * 12),
-        )
-        constraints.addConstraint("gripperPose", placement_constraint)
-
-        # add torque constraint
-        """
-        torque_residual = crocoddyl.ResidualModelJointEffort(
-            self.state,
-            self.actuation,
-            np.array([0] * 6),  # np.array([0] * 6)
-        )
-        torque_constraint = crocoddyl.ConstraintModelResidual(
-            self.state,
-            torque_residual,
-            np.array([-87] * 4 + [-12] * 3),
-            np.array([87] * 4 + [12] * 3),
-        )
-        constraints.addConstraint("JointsEfforts", torque_constraint)
-
-        # add velocities constraints
-
-        vref = pin.Motion.Zero()
+import numpy as np
+import matplotlib.pyplot as plt
 
-        joint_vel_residual = crocoddyl.ResidualModelFrameVelocity(
-            self.state,
-            self.robot.model.getFrameId(self.last_joint_name),
-            vref,
-            pin.LOCAL,
-        )
-        joint_vel_constraint = crocoddyl.ConstraintModelResidual(
-            self.state,
-            joint_vel_residual,
-            np.array([0] * self.nv),
-            np.array([1e-3] * self.nv),
-        )
-        constraints.addConstraint("shoulder_lift_joint_velocity", joint_vel_constraint)
-        
-        for joint_name in self.robot.model.names:
-            joint_vel_residual = crocoddyl.ResidualModelFrameVelocity(
-                self.state,
-                self.robot.model.getFrameId(joint_name),
-                vref,
-                pin.LOCAL,
-            )
-            joint_vel_constraint = crocoddyl.ConstraintModelResidual(
-                self.state,
-                joint_vel_residual,
-                np.array([-1e-6] * self.nv),
-                np.array([1e-6] * self.nv),
-            )
-            constraints.addConstraint(f"{joint_name}_velocity", joint_vel_constraint)
+from mim_robots.pybullet.env import BulletEnvWithGround
+import pinocchio as pin
+import mpc_utils
+import pin_utils
```

Comment:
make sure it's a local import: `import .pin_utils`
It helps the one reading the code to understand it's not a dependency but inside your own code.

Actually even better you call your own package like this:
`from agimus_controller import pin_utils`